xonsh/environ.py

```diff
@@ -865,6 +862,16 @@ class GeneralSetting(Xettings):
         "If enabled, the CommandsCache is saved between runs and can reduce the startup time.",
     )
 
+    ENABLE_COMMANDS_CACHE = Var(
+        default=True,
+        doc="command names in a directory are cached when enabled."
+        " It makes use of directory's modified time. "
+        "On some platforms it may not be accurate enough"
+        " (e.g. Windows, Linux save mtime in seconds). "
+        "setting it to False would disable the caching mechanism"
+        " and may slow down the shell",
+    )
```

Comment:
it is good to define `doc_default` as well. It will appear in the docs generated

Reply:
do you mean like this?

Reply:
I've updated it.